### PR TITLE
Fix deprecation warnings for FFmpeg 4.4

### PR DESCRIPTION
**Pull request by uklotzde**
https://pkgs.rpmfusion.org/cgit/free/mixxx.git/commit/?h=f34&id=0906bbdf0e5bd7ec4b1fccd583face042f24a056

### CODE REVIEW

File: src/sources/soundsourceffmpeg.cpp [hunk 1]
```diff
@@ -1020,9 +1020,15 @@ ReadableSampleFrames SoundSourceFFmpeg::readSampleFramesClamped(
     CSAMPLE* pOutputSampleBuffer = writableSampleFrames.writableData();
 
     AVPacket avPacket;
```
Comment (daschuer) [inline src/sources/soundsourceffmpeg.cpp:1022]: This does not fix the issue, that in future we must not rely on the size of AVPacket. 
It is intended to use av_packet_alloc() instead. 

see:

> Once removed, sizeof(AVPacket) will stop being a part of the public ABI.

**Review by daschuer — COMMENTED** [review]
Thank you. Did you consider to fix the upcoming incompatibility in turn?

File: src/sources/soundsourceffmpeg.cpp (hunk 1 above)
Comment (uklotzde) [reply src/sources/soundsourceffmpeg.cpp:1022]: Thanks for the hint.

**Comment by uklotzde** [comment]
I won't fix `EncoderFfmpegCore` because it is not used. This unmaintained class should be deleted.

**Pull request merged**